dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -484,11 +487,90 @@ private String getQuerySelect( TrackedEntityInstanceQueryParams params, boolean
         {
             select.append( ", TEI.trackedentityinstanceid AS teiid" );
         }
+
+        select.append( getOrderParamForSelectClause( params, select ) );
+
         select.append( SPACE );
 
         return select.toString();
     }
 
+    /**
+     * Gets all the static fields from the order parameters and appends them to
+     * the string builder, so they can be used in the query select clause
+     *
+     * @param params
+     * @param select current select statement
+     * @return all the static fields that are not yet present in the select
+     *         statement
+     */
+    private String getOrderParamForSelectClause( TrackedEntityInstanceQueryParams params, StringBuilder select )
+    {
+        StringBuilder orderQuery = new StringBuilder();
+
+        for ( OrderParam orderParam : params.getOrders() )
+        {
+            if ( isStaticColumn( orderParam.getField() ) )
+            {
+                if ( ENROLLED_AT.isPropertyEqualTo( orderParam.getField() ) )
+                {
+                    //In the main query, we need to use the main query alias to fetch the value of the field enrolledAt
```

Comment:
That should be clear from you now using a constant `MAIN_QUERY_ALIAS`. I don't think the comment is needed.